yt/data_objects/static_output.py

```diff
@@ -1119,27 +1120,46 @@ def relative_refinement(self, l0, l1):
         return self.refine_by ** (l1 - l0)
 
     def _assign_unit_system(self, unit_system):
-        if unit_system == "cgs":
-            current_mks_unit = None
-        else:
-            current_mks_unit = "A"
-        magnetic_unit = getattr(self, "magnetic_unit", None)
-        if magnetic_unit is not None:
-            if unit_system == "mks":
-                if current_mks not in self.magnetic_unit.units.dimensions.free_symbols:
-                    self.magnetic_unit = self.magnetic_unit.to("gauss").to("T")
-                self.unit_registry.modify("code_magnetic", self.magnetic_unit.value)
+        # we need to determine if the requested unit system
+        # is mks-like: i.e., it has a current with the same
+        # dimensions as amperes.
+        mks_system = False
+        if unit_system != "code":
+            # if the unit system is known, we can check if it
+            # has a "current_mks" unit
+            us = unit_system_registry[str(unit_system).lower()]
+            mks_system = us.base_units[current_mks] is not None
+        elif self._use_mks_em_units:
+            # we haven't specified the code unit system yet, but
+            # _use_mks_em_units tells us it's MKS
+            mks_system = True
+        # Now we get to the tricky part. If we have an MKS-like system but
+        # we asked for a conversion to something CGS-like, or vice-versa,
+        # we have to convert the magnetic field
+        if getattr(self, "magnetic_unit", None):
+            mag_dims = self.magnetic_unit.units.dimensions.free_symbols
+            if mks_system:
+                if current_mks not in mag_dims:
+                    self.magnetic_unit = self.quan(
+                        self.magnetic_unit.to_value("gauss") * 1.0e-4, "T"
+                    )
+                    # The following modification ensures that we get the conversion to
+                    # mks correct
+                    self.unit_registry.modify(
+                        "code_magnetic", self.magnetic_unit.value * 1.0e3 * 0.1**-0.5
+                    )
             else:
-                # if the magnetic unit is in T, we need to create the code unit
-                # system as an MKS-like system
-                if current_mks in self.magnetic_unit.units.dimensions.free_symbols:
-                    self.magnetic_unit = self.magnetic_unit.to("T").to("gauss")
-                # The following modification ensures that we get the conversion to
-                # cgs correct
-                self.unit_registry.modify(
-                    "code_magnetic", self.magnetic_unit.value * 0.1**0.5
-                )
-
+                if current_mks in mag_dims:
```

Comment:
```python
else:
    if ...:
        ...
```
Looks like this could be simplified into
```python
elif:
    ...
```

Reply:
Agreed.